DllImportGenerator/DllImportGenerator/StubCodeGenerator.cs

```diff
@@ -340,20 +340,22 @@ public override (string managed, string native) GetIdentifiers(TypePositionInfo
             var codeBlock = Block(UnsafeStatement(Block(allStatements)));
 
             // Define P/Invoke declaration
-            var dllImport = MethodDeclaration(retMarshaller.Generator.AsNativeType(retMarshaller.TypeInfo), dllImportName)
+            var dllImport = LocalFunctionStatement(retMarshaller.Generator.AsNativeType(retMarshaller.TypeInfo), dllImportName)
```

Comment:
My only question here would be how does this align with our desire to use this as a library? At first glance it seems this makes the stub generator even more tightly coupled to the `DllImport` scenario &ndash; which actually seems fine. Does this have any influence on the overall design and how it may expand?

Reply:
This should still enable our intent to use as a library. This change actually makes the whole `DllImport` scenario contained within `StubCodeGenerator.cs`.

The main question we need to answer is how we want to make this library extensible.

Do we want to enable people to provide their own set of default marshalling rules but still generate the internal stub code the same as StubCodeGenerator? Then we'll want to refactor MarshallingGenerator.Create to be an interface method, provide a default class implementation with our defaults, and pass it down to StubCodeGenerator.

Do we want to enable people to use the default generators but change the body of the function while preserving some parts (like the target dllimport function declaration)? Then we can extract this declaration along with other self-contained phases to separate protected functions that users can call when they're building their code-gen. For example, if someone wanted to build a generator that's basically identical to ours but used function pointers, they could override the code-gen to not produce this local function declaration and change the invoke stage to instead call their function pointer.

Do we want to enable people to build source generators for completely different scenarios like struct or COM-style marshalling? Then we just need a solidified design for our marshallers and the base StubCodeContext type and everyone else builds on top of those.

Reply:
>Do we want to enable people to build source generators for completely different scenarios like struct or COM-style marshalling? Then we just need a solidified design for our marshallers and the base StubCodeContext type and everyone else builds on top of those.

Yes.